test/e2e-go/features/accountPerf/sixMillion_test.go

```
@@ -0,0 +1,1383 @@
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/algorand/go-deadlock"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	clientApi "github.com/algorand/go-algorand/daemon/algod/api/client"
+	"github.com/algorand/go-algorand/daemon/algod/api/server/v2/generated"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/data/transactions/logic"
+	"github.com/algorand/go-algorand/libgoal"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/test/framework/fixtures"
+	"github.com/algorand/go-algorand/test/partitiontest"
+)
+
+// uses numberOfThreads to perform different operations (signing and preparing transactions) in parallel
+const numberOfThreads = 256
+
+// the frequency of printing progress status
+const printFreequency = 20
+
+// send transactions in groups or one by one
+const groupTransactions = true
+
+// number of elements queued in channels
+const channelDepth = 100
+
+// the test in intended for 6M apps/assets and 6K accounts. These variable values can be changed to modify this.
+const sixMillion = 100
+const sixThousand = 10
+
+// report additional information
+const verbose = false
+
+// transaction group size obtained from consensus parameter
+var maxTxGroupSize int
+
+type psKey struct {
+	sk *crypto.SignatureSecrets
+	pk basics.Address
+}
+
+type txnKey struct {
+	sk *crypto.SignatureSecrets
+	tx transactions.Transaction
+}
+
+// started as a goroutine which will listen to signed transactions and broadcast them
+func broadcastTransactions(queueWg *sync.WaitGroup, c libgoal.Client, sigTxnChan <-chan *transactions.SignedTxn, errChan chan<- error) {
+	for stxn := range sigTxnChan {
+		if stxn == nil {
+			break
+		}
+		var err error
+		for x := 0; x < 50; x++ { // retry only 50 times
+			_, err = c.BroadcastTransaction(*stxn)
```

It's ok to retry, but you should make sure that you test for expected error - not all errors are equal, and you might want to fail the test right away if the "unexpected" error is returned.